optuna/samplers/_tpe/parzen_estimator.py

```diff
@@ -200,8 +186,10 @@ def log_pdf(self, samples_dict: Dict[str, np.ndarray]) -> np.ndarray:
                     )
                     log_pdf = np.log(cdf + EPS) - np.log(p_accept + EPS)
             component_log_pdf += log_pdf
-        ret = special.logsumexp(component_log_pdf + np.log(self._weights), axis=1)
-        return ret
+        weighted_log_pdf = component_log_pdf + np.log(self._weights)
+        max_ = weighted_log_pdf.max(axis=1)
+        with np.errstate(divide="ignore"):
```

Comment:
Why was this necessary, when there doesn't seem to be a case where division by zero is involved?

Reply:
It follows the SciPy's implementation. I will add a comment why `np.errstate` is necessary. https://github.com/scipy/scipy/blob/7076a90ff42a2349c621e1a081fbf5f715c421c0/scipy/special/_logsumexp.py#L112-L113

Reply:
Is there a case where that would incur a division by zero?
(Except when `self._weights` are all zero, which would be invalid anyways)

Reply:
It introduced to resolve inf related bug. It never occurs in our situation, so we can remove `np.errstate`. Thank you for careful review!
https://github.com/scipy/scipy/issues/4014
https://github.com/scipy/scipy/pull/4016

Reply:
By the way, `np.errstate(under='ignore')` may be preferred. #2639